lemur/plugins/lemur_acme/challenge_types.py

```diff
@@ -213,39 +218,40 @@ def create_certificate(self, csr, issuer_options):
                 raise InvalidConfiguration(error)
         else:
             dns_provider = {}
-            dns_provider_options = None
             account_number = None
             provider_type = None
 
-        domains = self.acme.get_domains(issuer_options)
+            for domain in domains:
+                self.acme.autodetect_dns_providers(domain)
+
+        # Create pending authorizations that we'll need to do the creation
+        dns_authorization = authorization_service.create(
+            account_number, domains, provider_type
+        )
+
         if not create_immediately:
-            # Create pending authorizations that we'll need to do the creation
-            dns_authorization = authorization_service.create(
-                account_number, domains, provider_type
-            )
             # Return id of the DNS Authorization
             return None, None, dns_authorization.id
 
-        authorizations = self.acme.get_authorizations(
-            acme_client,
-            account_number,
-            domains,
-            dns_provider_plugin,
-            dns_provider_options,
-        )
-        self.acme.finalize_authorizations(
-            acme_client,
-            account_number,
-            dns_provider_plugin,
-            authorizations,
-            dns_provider_options,
-        )
-        pem_certificate, pem_certificate_chain = self.acme.request_certificate(
-            acme_client, authorizations, csr
+        pem_certificate, pem_certificate_chain = self.create_immediately(
+            acme_client, dns_authorization, csr
         )
         # TODO add external ID (if possible)
         return pem_certificate, pem_certificate_chain, None
 
+    @retry(stop_max_attempt_number=ACME_ADDITIONAL_ATTEMPTS, wait_fixed=5000)
+    def create_immediately(self, acme_client, order_info, csr):
```

Comment:
Since there's no pending certificate as we're issuing immediately, should we be sending metrics on attempts sort of like pending certs do?

Reply:
Added!